src/v/cluster/inventory_service.cc

```diff
@@ -100,12 +101,21 @@ default_leaders_provider::default_leaders_provider(
 ss::future<absl::node_hash_set<model::ntp>>
 default_leaders_provider::ntps(ss::abort_source& as) {
     absl::node_hash_set<model::ntp> ntps;
+    const auto self_node_id = config::node().node_id();
+    if (!self_node_id.has_value()) {
+        vlog(cst_log.warn, "node has no id, cannot find leadership");
+        co_return ntps;
+    }
+
     for (auto retry : std::ranges::iota_view{0, partition_leaders_retry}) {
         std::exception_ptr ep;
         try {
             co_await _leaders_table.local().for_each_leader(
-              [&ntps](auto tp_ns, auto pid, auto, auto) mutable {
-                  ntps.insert(model::ntp{tp_ns.ns, tp_ns.tp, pid});
+              [&ntps,
+               self_node_id](auto tp_ns, auto pid, auto node_id, auto) mutable {
+                  if (node_id == self_node_id) {
+                      ntps.insert(model::ntp{tp_ns.ns, tp_ns.tp, pid});
+                  }
```

Comment:
Just checking, we don't need to filter by core either, right? Inventory parsing is just going to happen on one core, is that right?

Reply:
Yes, here we just need to collect every NTP which has a leader on any shard of the current node. This code will always run on shard 0.